consensus/src/dag/dag_store.rs

```diff
@@ -142,6 +154,38 @@ impl Dag {
         Ok(())
     }
 
+    pub fn update_votes(&mut self, node: &Node, update_link_power: bool) {
+        if node.round() <= self.lowest_round() {
+            return;
+        }
+
+        for parent in node.parents_metadata() {
+            let voting_power = self
+                .epoch_state
+                .verifier
+                .get_voting_power(node.author())
+                .expect("must exist");
+            let node_status = self
+                .get_node_ref_mut(parent.round(), parent.author())
+                .expect("must exist");
```

Comment:
just add another expect here instead of unreachable below?

Reply:
we need a mutable reference to node_status. If i expect, then I wont get a reference to the Option in BTreeMap.